source/slang/glsl.meta.slang

```diff
@@ -407,13 +422,707 @@ ${{{{
 }
 }}}}
 
-[ForceInline] public int findLSB(int v) { return firstbitlow(v); }
-[ForceInline] public uint findLSB(uint v) { return firstbitlow(v); }
-[ForceInline] public vector<int,N> findLSB<let N:int>(vector<int,N> value)
+//
+// Section 8.1. Angle and Trigonometry Functions
+//
+
+__generic<T : __BuiltinFloatingPointType>
+__target_intrinsic(glsl)
+[__readNone]
+public T atan(T y, T x)
+{
+    return atan2(y, x);
+}
+
+__generic<T : __BuiltinFloatingPointType, let N : int>
+__target_intrinsic(glsl)
+[__readNone]
+public vector<T,N> atan(vector<T,N> y, vector<T,N> x)
+{
+    return atan2(y, x);
+}
+
+__generic<T : __BuiltinFloatingPointType>
+__target_intrinsic(glsl)
+__target_intrinsic(cuda, "$P_asinh($0)")
+__target_intrinsic(cpp, "$P_asinh($0)")
+[__readNone]
+public T asinh(T x)
+{
+    return log(x + sqrt(x * x + T(1)));
+}
+
+__generic<T : __BuiltinFloatingPointType, let N : int>
+__target_intrinsic(glsl)
+[__readNone]
+public vector<T,N> asinh(vector<T,N> x)
+{
+    VECTOR_MAP_UNARY(T, N, asinh, x);
+}
+
+__generic<T : __BuiltinFloatingPointType>
+__target_intrinsic(glsl)
+__target_intrinsic(cuda, "$P_acosh($0)")
+__target_intrinsic(cpp, "$P_acosh($0)")
+[__readNone]
+public T acosh(T x)
+{
+    return log(x + sqrt( x * x - T(1)));
+}
+
+__generic<T : __BuiltinFloatingPointType, let N : int>
+__target_intrinsic(glsl)
+[__readNone]
+public vector<T,N> acosh(vector<T,N> x)
+{
+    VECTOR_MAP_UNARY(T, N, acosh, x);
+}
+
+__generic<T : __BuiltinFloatingPointType>
+__target_intrinsic(glsl)
+__target_intrinsic(cuda, "$P_atanh($0)")
+__target_intrinsic(cpp, "$P_atanh($0)")
+[__readNone]
+public T atanh(T x)
+{
+    return T(0.5) * log((T(1) + x) / (T(1) - x));
+}
+
+__generic<T : __BuiltinFloatingPointType, let N : int>
+__target_intrinsic(glsl)
+[__readNone]
+public vector<T,N> atanh(vector<T,N> x)
+{
+    VECTOR_MAP_UNARY(T, N, atanh, x);
+}
+
+//
+// Section 8.2. Exponential Functions
+//
+
+__generic<T : __BuiltinFloatingPointType>
+__target_intrinsic(glsl)
+[__readNone]
+public T inversesqrt(T x)
+{
+    return rsqrt(x);
+}
+
+__generic<T : __BuiltinFloatingPointType, let N : int>
+__target_intrinsic(glsl)
+[__readNone]
+public vector<T, N> inversesqrt(vector<T, N> x)
+{
+    return rsqrt(x);
+}
+
+//
+// Section 8.3. Common Functions
+//
+
+__generic<T : __BuiltinFloatingPointType>
+__target_intrinsic(glsl)
+[__readNone]
```

Comment:
These functions should probably be marked [ForceInline] so slang can produce clean inlined code.